python/cudf/cudf/utils/gpu_utils.py

```diff
@@ -0,0 +1,96 @@
+def validate_setup():
+    from cudf._cuda.gpu import (
+        getDeviceCount,
+        driverGetVersion,
+        runtimeGetVersion,
+        getDeviceAttribute,
+        CudaDeviceAttr,
+    )
+    import warnings
+
+    gpus_count = getDeviceCount()
+
+    if gpus_count > 0:
+        # Cupy throws RunTimeException to get GPU count,
+        # hence obtaining GPU count by in-house cpp api above
+
+        # 75 - Indicates to get "cudaDevAttrComputeCapabilityMajor" attribute
+        # 0 - Get GPU 0
+        major_version = getDeviceAttribute(
+            CudaDeviceAttr.cudaDevAttrComputeCapabilityMajor, 0
+        )
+
+        if major_version >= 6:
+            # You have a GPU with NVIDIA Pascal™ architecture or better
+            # Hardware Generation	Compute Capability
+            #    Turing	                7.5
+            #    Volta	                7.x
+            #    Pascal	                6.x
+            #    Maxwell	            5.x
+            #    Kepler	                3.x
+            #    Fermi	                2.x
+            pass
+        else:
+
+            warnings.warn(
+                "You will need a GPU with NVIDIA Pascal™ architecture or \
+                    better"
```

Comment:
Should include in the warning the GPU name / compute capability detected. We've had issues where someone has something like a GT 710 as GPU 0 and a Titan RTX as GPU 1 and they were unclear why they had issues.